samples/example_vptr/example_vptr.cpp

```diff
@@ -0,0 +1,128 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2016 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  example_sycl_application.cpp
+ *
+ *  Description:
+ *    Sample code that walks through the basics of executing matrix addition
+ *    using the virtual pointer in SYCL.
+ *
+ **************************************************************************/
+
+#include <CL/sycl.hpp>
+
+#include <iostream>
+
+#include "pointer_alias.hpp"
+#include "virtual_ptr.hpp"
+
+using namespace cl::sycl;
+
+const size_t N = 100;
+const size_t M = 150;
+
+/* This sample creates three device-only arrays, then initialises them
+ * on the device. After that, it adds two of them together, storing the
+ * result in the third buffer. It then verifies the result of the kernel
+ * on the host by using a host accessor to gain access to the data. */
+int main() {
+  /* Destroying SYCL objects like queues and buffers blocks until all work
```

Comment:
Yes, I copied code from another class - I will go over all comments in this one and tidy them up.